### PR TITLE
Add disk encryption options documentation

### PR DESCRIPTION
**Pull request by bryan-cox**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This PR reorganizes the disk pages into their own section and adds a new documentation section discussing disk encryption options.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
N/A

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [x] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Reorganized the disk pages into their own section and added a new documentation section discussing disk encryption options.
```

### CODE REVIEW

**Comment by bryan-cox** [comment]
Happy to squash those two commits into one. Just left them separate for now for peer review.

File: docs/book/src/topics/disk-encryption.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Disk Encryption
+This document describes how to configure different encryption options for disks allocated to VMs provisioned in Azure. 
+
+## Azure Disk Storage Server-Side Encryption
+Azure Disk Storage Server-Side Encryption (SSE) is also referred to as encryption-at-rest. This encryption option does not encrypt temporary disks or disk caches.
+
+When enabled, Azure Disk Storage SSE encrypts data stored on Azure managed disks, i.e. OS and data disks. This option can be enabled using customer managed keys.
+
+Customer managed keys must be configured through a Disk Encryption Set (DES) resource. For more information on Azure Disk Storage SSE, please see this [link](https://learn.microsoft.com/en-us/azure/virtual-machines/disk-encryption).
```
Comment (mboersma) [inline docs/book/src/topics/disk-encryption.md:9]:
```suggestion
Customer-managed keys must be configured through a Disk Encryption Set (DES) resource. For more information on Azure Disk Storage SSE, please see this [link](https://learn.microsoft.com/azure/virtual-machines/disk-encryption).
```
This is nitpicky, but I try to remove the `en-us/` part of doc URLs so the user can access a localized version depending on their language preferences.

File: docs/book/src/topics/disk-encryption.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+### Example with OS Disk using DES
+When using customer managed keys, you only need to provide the DES ID within the managedDisk spec.
```
Comment (mboersma) [inline docs/book/src/topics/disk-encryption.md:12]:
```suggestion
When using customer-managed keys, you only need to provide the DES ID within the managedDisk spec. 
```

**Review by mboersma — COMMENTED** [review]
This is great @bryan-cox, thank you! I just had a few nitpicks.

File: docs/book/src/topics/disk-encryption.md (hunk 1 above)
Comment (bryan-cox) [reply docs/book/src/topics/disk-encryption.md:9]: That makes total sense. I didn't even think about that. I'll try to remember that for next time :)

**Comment by bryan-cox** [comment]
@mboersma -Thanks for the feedback! I've addressed all the comments. Just let me know when you think it's good to go and I'll squash them all into one commit.

**Review by mboersma — COMMENTED** [review]
/lgtm

Squash whenever time permits and I'm happy to approve this PR.

**Comment by bryan-cox** [comment]
Commits are all squashed now.

**Review by mboersma — APPROVED** [review]
/approve

**Pull request merged**